release_process.md

```diff
@@ -1,11 +1,11 @@
 # Release process
 
-There are five total packages we release. Three are the `lancedb` packages
-for Python, Rust, and Node.js. The other two are the legacy `vectordb`
+There are six total packages we release. Four are the `lancedb` packages
+for Python, Rust, Java, and Node.js. The other two are the legacy `vectordb`
 packages for Rust and node.js.
```

Comment:
While we are updating this, we can remove the `vectordb` Rust package, which hasn't been part of our release process for a little while.
```suggestion
There are five total packages we release. Four are the `lancedb` packages
for Python, Rust, Java, and Node.js. The other one is the legacy `vectordb`
package node.js.
```